force-app/main/default/classes/DMLOperations.cls

```diff
@@ -180,15 +235,21 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
-
-		// Get the account based on the name or create one if it doesn't exist
-
-		// Link the Contact to its Account using the Account Id
-
-		// Upsert the Contacts
+		List<Contact> upsertContactsList = new List<Contact>();
+        // Loop through each Contact
+        for(Contact c : contactsToUpsert) {
+            // Extract the Account Name from Contact's LastName
+            String lastName = c.LastName;
+            // Get the account based on the name or create one if it doesn't exist
+            Account a = upsertAccount(lastName);
+            // Link the Contact to its Account using the Account Id
+            c.AccountId = a.Id;
+            upsertContactsList.add(c);
```

Comment:
Same thing here with `a` — use `acc` or `acct`:
```suggestion
            Account acc = upsertAccount(lastName);
            // Link the Contact to its Account using the Account Id
            con.AccountId = acc.Id;
            upsertContactsList.add(con);
```
This may seem trivial, but it's important that you choose variable names that tell us exactly what they represent at a glance. Single letters will never meet that requirement.